app-libs/exchange-oracle/src/exchange_rate_oracle.rs

```diff
@@ -66,8 +68,10 @@ where
 		self.metrics_exporter.increment_number_requests(source_id.clone());
 
 		let base_url = self.oracle_source.base_url()?;
-		let http_client = HttpClient::new(true, self.oracle_source.request_timeout(), None, None);
+		let http_client =
+			HttpClient::new(true, self.oracle_source.request_timeout(), None, None, true);
 		let mut rest_client = RestClient::new(http_client, base_url.clone());
+		rest_client.set_root_cert(self.oracle_source.root_certificate_content());
 
```

Comment:
http_client::authenticated_connection = true -> Tls connection is only established if the certificate supplied by the oracle_source matches the server's root certificate for the oracle_source.base_url